tests/php/validation/test-class-amp-validation-error-taxonomy.php

```diff
@@ -1130,6 +1130,75 @@ public function test_filter_manage_custom_columns() {
 		$this->assertContains( 'CSS', $filtered_content );
 	}
 
+	/**
+	 * Gets the test data for test_filter_manage_custom_columns_error_appears().
+	 *
+	 * @return array An associative array of the test data.
+	 */
+	public function get_filter_manage_custom_columns_data() {
+		return [
+			'json_error_syntax'              => [
+				AMP_Tag_And_Attribute_Sanitizer::JSON_ERROR_SYNTAX,
+				null,
+				'Syntax error',
+			],
+			'json_error_utf8'                => [
+				AMP_Tag_And_Attribute_Sanitizer::JSON_ERROR_UTF8,
+				null,
+				'Malformed UTF-8 characters, possibly incorrectly encoded',
+			],
+			'json_error_inf_or_nan'          => [
+				AMP_Tag_And_Attribute_Sanitizer::JSON_ERROR_INF_OR_NAN,
+				null,
+				'One or more NAN or INF values in the value to be encoded',
+			],
+			'json_error_empty'               => [
+				AMP_Tag_And_Attribute_Sanitizer::JSON_ERROR_EMPTY,
+				null,
+				'Expected JSON, got an empty value',
+			],
+			'error_message_present_in_error' => [
+				AMP_Style_Sanitizer::STYLESHEET_FETCH_ERROR,
+				'The stylesheet could not be found',
+				'The stylesheet could not be found',
+			],
+		];
+	}
+
+	/**
+	 * Test the error message behavior of filter_manage_custom_columns.
+	 *
+	 * @dataProvider get_filter_manage_custom_columns_data
+	 * @covers \AMP_Validation_Error_Taxonomy::filter_manage_custom_columns()
+	 *
+	 * @param string      $error_code             The error code in the validation error.
+	 * @param string|null $error_message          The error message in the validation error, if any.
+	 * @param string      $expected_error_message The error message that should appear in the custom column.
+	 */
+	public function test_filter_manage_custom_columns_error_appears( $error_code, $error_message, $expected_error_message ) {
+		$this->accept_sanitization_by_default( false );
+		AMP_Validation_Error_Taxonomy::register();
+		$validation_error = [ 'code' => $error_code ];
+
+		if ( $error_message ) {
+			$validation_error['message'] = $error_message;
+		}
+
+		$initial_content = 'here is the initial content';
+		$term_id         = self::factory()->term->create(
+			[
+				'taxonomy'    => AMP_Validation_Error_Taxonomy::TAXONOMY_SLUG,
+				'description' => wp_json_encode( $validation_error ),
+			]
+		);
+
+		$GLOBALS['pagenow'] = 'post.php';
+		$filtered_content   = AMP_Validation_Error_Taxonomy::filter_manage_custom_columns( $initial_content, 'error_code', $term_id );
+
+		$this->assertStringStartsWith( $initial_content . '<button type="button" aria-label="Toggle error details"', $filtered_content );
+		$this->assertStringContainsString( $expected_error_message, $filtered_content );
```

Comment:
This method is causing test failures:

> `PHP Fatal error:  Call to undefined method Test_AMP_Validation_Error_Taxonomy::assertStringContainsString() in /tmp/wordpress/src/wp-content/plugins/amp/tests/php/validation/test-class-amp-validation-error-taxonomy.php on line 1199`

Reply:
Sorry for missing that. https://github.com/ampproject/amp-wp/pull/4340/commits/ae5df635383dc7d2304b0564574683d3ad42871a fixes it.